### PR TITLE
Restricted user acces to selected team members

### PR DESCRIPTION
**Pull request by venugopal1234567**
Restricting users to send TODO to only for the selected team members.

### CODE REVIEW

File: webapp/src/actions.js [hunk 1]
```diff
@@ -129,9 +131,19 @@ export const bump = (id) => async (dispatch, getState) => {
...
-    return async (doDispatch) => {
-        const {data} = await doDispatch(UserActions.autocompleteUsers(username));
-        return data.users;
+    return async (doDispatch, getState) => {
+        let userIds = [];
+        let users = [];
+        let team = TeamSelector.getCurrentTeam(getState())
+        const teamMembers = await doDispatch(TeamActions.getTeamMembers(team.id));
+        for( let j =0; j< teamMembers.data.length; j++) {
+            const user = await doDispatch(UserActions.getUser(teamMembers.data[j].user_id));
```
Comment (mickmister) [inline webapp/src/actions.js:140]: It seems we've changed the expected functionality, and made this a bit unperformant.

Originally:

- We used the `autocompleteUsers` to search for a user by username

Now:

- We fetch all team members of the current team from the API
- Then individually fetch each user of that team from the API

It looks like the `autocompleteUsers` function accepts a `teamId` argument:

https://github.com/mattermost/mattermost-webapp/blob/b506465285bd4e95d2242349046ebb9cc95e6385/packages/mattermost-redux/src/actions/users.ts#L908

Can we use that existing function instead of introducing the implementation used below?

File: webapp/src/actions.js [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
     GET_IN_ISSUES,
+    GET_ISSUES,
     GET_OUT_ISSUES,
-    UPDATE_RHS_STATE,
-    SET_RHS_VISIBLE,
+    OPEN_ROOT_MODAL,
+    RECEIVED_SHOW_RHS_ACTION,
     SET_HIDE_TEAM_SIDEBAR_BUTTONS,
+    SET_RHS_VISIBLE,
+    UPDATE_RHS_STATE,
```
Comment (mickmister) [inline webapp/src/actions.js:14]: Can we revert these changes, as they are unrelated to the purpose of this PR?

**Review by mickmister — CHANGES_REQUESTED** [review]
Thanks for this improvement @venugopal1234567 :+1:

Looks mostly good, though I have a request to use the existing `autocompleteUsers` function that accepts a team id as an optional parameter, rather than introducing new autocomplete filtering logic

File: webapp/src/actions.js [hunk 3]
```diff
@@ -129,9 +130,17 @@ export const bump = (id) => async (dispatch, getState) => {
...
+        const users = [];
+        const team = TeamSelector.getCurrentTeam(getState());
+        const {data} = await doDispatch(UserActions.autocompleteUsers(username, team.id));
+        data.users.forEach((user) => {
+            //Returning only non deleted users
+            if (user.delete_at === 0) {
+                users.push(user);
+            }
+        });
+        return users;
```
Comment (mickmister) [inline webapp/src/actions.js:143]: This can be done with `filter`:

```suggestion
        return data.users.filter((user) => user.delete_at === 0);
```

**Review by mickmister — APPROVED** [review]
LGTM! Thank you @venugopal1234567!

**Review by larkox — APPROVED** [review]

**Comment by mickmister** [comment]
@DHaussermann this PR has approved by dev review. Wondering your thoughts on testing this post merge? The feature has very small scope and low risk for regression.

This change is using a supported optional `team_id` parameter to the user autocomplete function exposed by the webapp. We were simply not using this parameter until now.

**Comment by mickmister** [comment]
@DHaussermann I'm wondering if we can merge this with my comment above in mind?
